src/modules/mavlink/streams/ESC_INFO.hpp

```diff
@@ -49,50 +51,126 @@ class MavlinkStreamESCInfo : public MavlinkStream
 
 	unsigned get_size() override
 	{
-		static constexpr unsigned size_per_batch = MAVLINK_MSG_ID_ESC_INFO_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES;
-		return _esc_status_sub.advertised() ? size_per_batch * _number_of_batches : 0;
+		static constexpr unsigned message_size = MAVLINK_MSG_ID_ESC_INFO_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES;
+		return _esc_status_subs.advertised_count() * message_size;
 	}
 
 private:
 	explicit MavlinkStreamESCInfo(Mavlink *mavlink) : MavlinkStream(mavlink) {}
 
-	uORB::Subscription _esc_status_sub{ORB_ID(esc_status)};
-	uint8_t _number_of_batches{0};
+	uORB::SubscriptionMultiArray<esc_status_s> _esc_status_subs{ORB_ID::esc_status};
+
+	static constexpr uint8_t MAX_ESC_OUTPUTS = 12; // See output_functions.hpp
+	static constexpr uint8_t ESCS_PER_MSG = MAVLINK_MSG_ESC_INFO_FIELD_TEMPERATURE_LEN;
+	static constexpr uint8_t MAX_NUM_MSGS = MAX_ESC_OUTPUTS / ESCS_PER_MSG;
+
+	static constexpr hrt_abstime ESC_TIMEOUT = 100000;
+
+	// From #include <mixer_module/output_functions.hpp>
+	// We don't include the header here as to not introduce intermodule dependencies
+	static constexpr int OUTPUT_FUNCTION_MOTOR1 = 101;
```

Comment:
Instead of the magic numbers put the actuator_function constants in the msg. https://github.com/PX4/PX4-Autopilot/blob/c3b0ed3c5a1ac9214720011987ad3077ece50da8/msg/EscReport.msg#L12